examples/show_isophotes_and_curvature.py

```diff
@@ -0,0 +1,141 @@
+def gaussian_curvature(data, resolutions=None, on=None):
+    if resolutions is not None:
+        q = gus.create.vertices.raster(
+            data.knot_vector_bounds.tolist(), resolutions
+        ).vertices
+        v1 = data.derivative(q, [1, 0])
+        v2 = data.derivative(q, [0, 1])
+        v3 = data.derivative(q, [2, 0])
+        v4 = data.derivative(q, [0, 2])
+        v5 = data.derivative(q, [1, 1])
+    elif on is not None:
+        v1 = data.derivative(on, [1, 0])
+        v2 = data.derivative(on, [0, 1])
+        v3 = data.derivative(on, [2, 0])
+        v4 = data.derivative(on, [0, 2])
+        v5 = data.derivative(on, [1, 1])
+    # Using notation from "An introduction to NURBS historical perspective"
+    normals = np.cross(v1, v2)
+    A = np.einsum("ij,ij->i", normals, v3)
+    B = np.einsum("ij,ij->i", normals, v4)
+    C = np.einsum("ij,ij->i", normals, v5)
+
+    k_g = (A * C - B**2) / np.linalg.norm(normals, axis=1) ** 4
+    return k_g
+
+
+# Mean Curvature
+def mean_curvature(data, resolutions=None, on=None):
+    if resolutions is not None:
+        q = gus.create.vertices.raster(
+            data.knot_vector_bounds.tolist(), resolutions
+        ).vertices
+        v1 = data.derivative(q, [1, 0])
+        v2 = data.derivative(q, [0, 1])
+        v3 = data.derivative(q, [2, 0])
+        v4 = data.derivative(q, [0, 2])
+        v5 = data.derivative(q, [1, 1])
+    elif on is not None:
+        v1 = data.derivative(on, [1, 0])
+        v2 = data.derivative(on, [0, 1])
+        v3 = data.derivative(on, [2, 0])
+        v4 = data.derivative(on, [0, 2])
+        v5 = data.derivative(on, [1, 1])
+    # Using notation from "An introduction to NURBS historical perspective"
+    normals = np.cross(v1, v2)
+    A = np.einsum("ij,ij->i", normals, v3)
+    B = np.einsum("ij,ij->i", normals, v4)
+    C = np.einsum("ij,ij->i", normals, v5)
+    v1v1 = np.einsum("ij,ij->i", v1, v1)
+    v1v2 = np.einsum("ij,ij->i", v1, v2)
+    v2v2 = np.einsum("ij,ij->i", v2, v2)
+
+    k_g = (A * v1v1 - 2 * v1v2 * B + C * v2v2) / (
+        2 * np.linalg.norm(normals, axis=1) ** 3
+    )
+    return k_g
+
+
+# Spline Adaptor to plot function
+isophotes = sp.SplineDataAdaptor(
+    a, function=isophotes_to_direction(normal=[0, 0, 1])
+)
+mean_curvature = sp.SplineDataAdaptor(a, function=mean_curvature)
+gaussian_curvature = sp.SplineDataAdaptor(a, function=gaussian_curvature)
+
+# Set Plot options
+a.spline_data["isophotes"] = isophotes
+a.spline_data["gaussian_curvature"] = gaussian_curvature
+a.spline_data["mean_curvature"] = mean_curvature
+a.show_options["lighting"] = "off"
+a.show_options["control_points"] = False
+a.show_options["knots"] = False
+
+# Show Isophotes
+a.show_options["data_name"] = "isophotes"
+a.show(n_colors=2, resolutions=1000, title="Isophotes")
+
+# Show Gaussian Curvature
+a.show_options["data_name"] = "gaussian_curvature"
+a.show_options["scalarbar"] = True
+a.show(n_colors=2, resolutions=100, title="Gaussian Curvature")
```

Comment:
what does `n_colors` do?

Reply:
Didn't have the expected effect. Deleted it